app/app_iot_hub_test.go

```diff
@@ -1284,3 +1285,110 @@ func TestSyncIoTHubDevices(t *testing.T) {
 		})
 	}
 }
+
+func TestVerifyDeviceTwinIotHub(t *testing.T) {
+	t.Parallel()
+	noLogger := log.NewEmpty()
+	noLogger.Logger.Out = io.Discard
+
+	// parse public key
+	pem_data, err := ioutil.ReadFile("../testdata/public.pem")
+	if err != nil {
+		panic(err)
+	}
+	pubkeyStr := string(pem_data)
+	var pubkey model.PublicKey
+	err = pubkey.UnmarshalText([]byte(pubkeyStr))
+	if err != nil {
+		panic(err)
+	}
+
+	type testCase struct {
+		Name string
+
+		Integration model.Integration
+		Req         model.PreauthRequest
+
+		DataStore func(t *testing.T, self *testCase) *storeMocks.DataStore
+		Hub       func(t *testing.T, self *testCase) *hubMocks.Client
+
+		Error error
+	}
+	testCases := []testCase{{
```

Comment:
There are no test cases covering expected errors.

Reply:
yup; I was trying to cut some corners to speed things up, but I guess it doesn't make sense anymore, so I'll add more tests there

Reply:
fixed